tide_landing_page.info.yml

```diff
@@ -17,6 +17,7 @@ dependencies:
   - dpc-sdp:tide_core
   - dpc-sdp:tide_event
   - dpc-sdp:tide_media
+  - dpc-sdp:tide_automated_listing
```

Comment:
@asarfara cc @anthony-malkoun you are introducing many new dependencies: tide_api, tide_search, and tide_automated_listing. Content type modules (landing page, page, etc.) should not depend on extra feature modules.

Reply:
This is also a circular dependency. tide_automated_list already depends on tide_landing_page.

Reply:
Hi @sonnykt I think tide_automated_listing from an independent module point of view should only create fields and not attach them to landing page automatically. It should be up to the consumer i.e. tide_landing_page to use this field type. Maybe in the future someone may not want to use tide_landing_page and would like to use tide_automated_listing module with tide_event only or some other content type, in that scenario your code will fail as it will be looking for tide_landing_page which is either disabled or removed. 

The owner of landing_page_component is not tide_automated_listing. You can see other implementation related to that in below code:

```
      $is_featured_news = ($bundle == 'featured_news' && $moduleHandler->moduleExists('tide_news'));
      $is_embedded_webform = ($bundle == 'embedded_webform' && $moduleHandler->moduleExists('tide_webform'));
      if ($is_featured_news || $is_embedded_webform) {
      $is_automated_listing = ($bundle == 'automated_card_listing' && $moduleHandler->moduleExists('tide_automated_listing'));

      if ($is_featured_news || $is_embedded_webform || $is_automated_listing) {
```

I have already removed circular dependency in this pr https://github.com/dpc-sdp/tide_automated_listing/pull/1/files

By just using this field type regardless of my updates, consumer has already agreed that they will be using tide_api and tide_search.

Reply:
Hi @asarfara cc @anthony-malkoun,

As my previous comment, Tide Automated Listing is built specifically for Tide Landing Page because landing page is the only default content type with the concept of "component". Tide Landing Page is not actually a hard dependency of Tide Automated Listing. You can safely remove `tide_landing_page` from the `info.yml` of tide_automated_listing without any issues. Tide Automated Listing is still able to add itself to landing page (if exists) via entity_bundle_create hook.

Forcing tide_landing_page to depend on its components is not logically correct IMHO. When you are building an independent component, none of the default content types should have that component is a dependency.